### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by cf2080**
# Description
Things Changed/Added:
1. Added Httpost login method and Login Request Class in User Controller
2. Added VerifyPassword method in Password Service class in order to verify whether the password inputted is correct
3. Added IncorrectPasswordCount attribute to User model class in order to keep track of password attempts so a user can be disabled
4. Added cors to webapiconfig class
5. Added a simple login page to the front just for testing purposes
## Type of change
- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- [ ] The code hasn't been unit tested but will add some soon

# Notes

Put any notes here

### CODE REVIEW

File: Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs [hunk 1]
```diff
@@ -3,10 +3,24 @@ namespace DataAccessLayer.Migrations
     using System;
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs]: Hmm, why was this old migration moved or changed? Migrations tend to be treated as historical artifacts

Comment (julianpoy) [reply Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs]: If you're trying to add a new field, make a new migration to modify the users table.

File: Backend/DataAccessLayer/Migrations/Configuration.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@ internal sealed class Configuration : DbMigrationsConfiguration<DataAccessLayer.
     {
         public Configuration()
         {
-            AutomaticMigrationsEnabled = false;
+            AutomaticMigrationsEnabled = true;
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/Configuration.cs]: Revert this unless we're making a policy change to have automatic migrations.

File: Backend/DataAccessLayer/Migrations/201903062125096_initial-create.resx [hunk 3]
```diff
@@ -118,7 +118,7 @@
...
...VlgxuiKE1kodfWxyJeLcditCJMl5bkld8l7W/TNulUmb32vlAJ9xpBxIstuyPxT3VRM6sqNc1zSsuKiRFj+shurH4I4EAUb9XDUkGTHsuBQyemV3rw54MYblyJoiQSZtxkn5gsyhNKsTRBduqZBqXX4ZuV5lWdlwGywq5VayzepppaaWEujvaxepKwtl5lx19ATkFjfo6SCh9wpVu6lSRnUpwoioeQKNfBZ6vDlcNVMnVYY8QDLUHiNXMsjj5IbbY+Wu3zxWbrg9Vu4CzmPlhjvIlUTjglANEVq7V+mYKzG8YlWVK69opa1sOA4XezHguujXwnrryQ5jusnjvWBy8VB7jMJLPI9UmOhgctFrvGBt0Uh7hOSpnYdIhjpIkT6kC4Kkg+1xig/lPFhxpjti/PitQ4xnOugr974tKC03vg1a+hSqx0xnt0DuNWD2tkPbKGdvBzn7DZj97dA2ytnvLmfTG7IQTxrWdIgP2auyEByy0ZdzsR30SqrkneUlGfcs/yzlmYMk53u+FVdJAuMlpoFKeqKuTgDtlVTgRZfZqf2ZjRgFfezpgjvC6QykisvVZu/svFfq4309PTVLSpe1a6wdveQecvo5BKpfonRGtfvu0uLiT0Q4CyK+88jy+93bVtqtdLe3p88dHBq3kH/cvR60L+BKeWhvEhd6XNVDqjzjx9yF5dD8M6K/MMa/f4ohToyJQHe5MM6Mv7asWdX3pl6epRYaRbtZak3zJz35nZo/jVJ1bvDshFRq4uyEVdeo2U35dc0XBJxG/ZftIKsNlt1OoqGJsh/Q3p7hDiFjf89we5Rxc+OC6q87ti2mtDvG135F7bmhERduj9luqBRst2ugbNW1aCjEHKpN8TJaE9VCcKvGw6a+Q/yKQBOf+ni6sS031fBruxKNTYk65PrC/iH7FXkR16W6Fs2Juq7GYVoS1ScgGk7uP5tos5LO1xD6H5wcnILJZGvGfOanlluSKF1SinZ3oIir66ECk0SCMV74ujAatcA/EhZGaeAU3DGfhCoIFW4ZvCkrVNS0B2ziH/VdijIPJkH097F9bAHFpDpGT/hVSJmbyX1bE5QbILRrvQUcj88SPVXBfJUh3fu8JVCiviwiPIIXMASTE26TJ9hGNjTZdzAnzip9yTeDPH8QRbUPrimZC+LJBGNNjz/Rhl1v+fo/oq7atLosAAA=</value>
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201903062125096_initial-create.resx]: Revert this

File: Backend/DataAccessLayer/Models/User.cs [hunk 4]
```diff
@@ -39,6 +39,10 @@ public User()
         public string SecurityQ2Answer { get; set; }
         public string SecurityQ3 { get; set; }
         public string SecurityQ3Answer { get; set; }
+
+        [Required]
+        public int IncorrectPasswordCount { get; set; }
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Models/User.cs]: If this is non-nullable, make sure the user constructor for this model sets a default value (0)

File: Backend/Dev/App.config [hunk 5]
```diff
@@ -1,11 +1,11 @@
...
 <configuration>
   <configSections>
-    <!-- For more information on Entity Framework configuration, visit http://go.microsoft.com/fwlink/?LinkID=237468 -->
+    
     <section name="entityFramework" type="System.Data.Entity.Internal.ConfigFile.EntityFrameworkSection, EntityFramework, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" requirePermission="false" />
-  </configSections>
+  <!-- For more information on Entity Framework configuration, visit http://go.microsoft.com/fwlink/?LinkID=237468 --></configSections>
   <startup>
-    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.6.2" />
+    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.6.1" />
```
Comment (julianpoy) [inline Backend/Dev/App.config]: No downgrade here. Please revert this file.

File: Backend/Dev/Dev.csproj [hunk 6]
```diff
@@ -8,10 +8,11 @@
     <OutputType>Exe</OutputType>
     <RootNamespace>Program</RootNamespace>
     <AssemblyName>Dev</AssemblyName>
-    <TargetFrameworkVersion>v4.6.2</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
```
Comment (julianpoy) [inline Backend/Dev/Dev.csproj]: Same as above

File: Backend/Dev/packages.config [hunk 7]
```diff
@@ -1,4 +1,4 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="EntityFramework" version="6.2.0" targetFramework="net462" />
+  <package id="EntityFramework" version="6.2.0" targetFramework="net461" />
```
Comment (julianpoy) [inline Backend/Dev/packages.config]: Same as above. I'm going to stop commenting on this issue, please check all other instances of this change.

File: Backend/ServiceLayer/Services/PasswordService.cs [hunk 8]
```diff
@@ -59,5 +59,18 @@ public string[] QueryPwnedApi(string prefix,string url)
...
+        /// Used to verify entered password from user INCOMPLETE
+        /// </summary>
+        /// <param name="enteredPassword"></param>
+        /// <param name="storedHash"></param>
+        /// <param name="storedSalt"></param>
+        /// <returns></returns>
+        public bool VerifyPassword(string password, string hash, byte[] salt)
+        {
+            var rfc2898DeriveBytes = new Rfc2898DeriveBytes(password, salt);
+            return Convert.ToBase64String(rfc2898DeriveBytes.GetBytes(256)) == hash;
```
Comment (julianpoy) [inline Backend/ServiceLayer/Services/PasswordService.cs]: Hmm - I believe this functionality exists somewhere.

File: Backend/UnitTesting/ITestingUtils.cs [hunk 9]
```diff
@@ -12,9 +12,9 @@ interface ITestingUtils
     {
         User createUser();
         Session createSession(User user);
-        Service createService(bool enabled);
-        Claim createClaim(User user, Service service);
+        //Service createService(bool enabled);
+        //Claim createClaim(User user, Service service);
         bool isEqual(string[] arr1, string[] arr2);
-        Client createClientObject();
+        //Client createClientObject();
```
Comment (julianpoy) [inline Backend/UnitTesting/ITestingUtils.cs]: What is the reason for these changes?

File: Backend/UnitTesting/UserServiceUT.cs [hunk 10]
```diff
@@ -381,7 +381,7 @@ public void Is_User_Manager_Over_Independent()
 
             using (_db = tu.CreateDataBaseContext())
             {
-                Assert.IsFalse(us.IsManagerOver(_db, unassociatedUser, subject));
+                //Assert.IsFalse(us.IsManagerOver(_db, unassociatedUser, subject));
```
Comment (julianpoy) [inline Backend/UnitTesting/UserServiceUT.cs]: Please remove these tests completely before PR. These shouldn't exist in the system anymore (applies to tests you commented below as well).

File: Backend/WebAPI/App_Start/WebApiConfig.cs [hunk 11]
```diff
@@ -10,6 +12,17 @@ public static class WebApiConfig
         public static void Register(HttpConfiguration config)
         {
             // Web API configuration and services
+            var cors = new EnableCorsAttribute("http://localhost:8080", "*", "*");
+            config.EnableCors(cors);
+
+
+            // Set JSON formatter as default one and remove XmlFormatter
+            var jsonFormatter = config.Formatters.JsonFormatter;
+            jsonFormatter.SerializerSettings.ContractResolver = new CamelCasePropertyNamesContractResolver();
```
Comment (julianpoy) [inline Backend/WebAPI/App_Start/WebApiConfig.cs]: :+1:

File: Frontend/src/router.js [hunk 12]
```diff
@@ -25,6 +27,16 @@ let router = new VueRouter({
...
     },
+    {
+      path: '/login',
+      name: 'login',
+      component: Login
+    },
+    {
+      path: '/portal',
+      name: 'portal',
+      component: Portal
```
Comment (julianpoy) [inline Frontend/src/router.js]: Let's call this dashboard instead of portal.

File: Frontend/src/views/Login.vue [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+                    password: ""
+                }
+            }
+        },
+        mounted(){
+
+        },
+        methods: {
+            login() {
+               axios.post('http://localhost:50803/api/users/login').then(input => {this.input = input.data; console.log("sent!");})
```
Comment (julianpoy) [inline Frontend/src/views/Login.vue]: This could use some better result/error handling.

**Review by julianpoy — CHANGES_REQUESTED** [review]

**Comment by julianpoy** [comment]
There are a giant number of bootstrap and other frontend files added to the backend project. Please remove those.

File: Backend/UnitTesting/UserServiceUT.cs (hunk 10 above)
Comment (jbcayton) [reply Backend/UnitTesting/UserServiceUT.cs]: Do you want the class deleted or just those specific tests?

File: Backend/DataAccessLayer/Database/dbContext.cs [hunk 14]
```diff
@@ -14,12 +14,10 @@ public class DatabaseContext : DbContext
     {
         public DatabaseContext()
         {
-            this.Database.Connection.ConnectionString = "Data Source=localhost;Initial Catalog=SSO;Integrated Security=True";
+            Database.Connection.ConnectionString = "Data Source=Localhost;Initial Catalog=SSO;Integrated Security=True";
```
Comment (alfredovargasjr) [inline Backend/DataAccessLayer/Database/dbContext.cs]: Why was this change made?

File: Backend/UnitTesting/ITestingUtils.cs (hunk 9 above)
Comment (alfredovargasjr) [reply Backend/UnitTesting/ITestingUtils.cs]: These are creating to project build errors, these classes do not existing in the solution

File: Backend/UnitTesting/UserServiceUT.cs (hunk 10 above)
Comment (alfredovargasjr) [reply Backend/UnitTesting/UserServiceUT.cs]: This has been fixed on the bug branch

File: Backend/WebAPI/Areas/HelpPage/Views/Web.config [hunk 15]
```diff
@@ -9,7 +9,7 @@
   </configSections>
 
   <system.web.webPages.razor>
-    <host factoryType="System.Web.Mvc.MvcWebRazorHostFactory, System.Web.Mvc, Version=5.2.4.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35" />
```
Comment (alfredovargasjr) [inline Backend/WebAPI/Areas/HelpPage/Views/Web.config]: This WebAPI project will be replaced by a clean Web Api project that gets rid of unnecessary MVC packages. Refer to bug branch on the SSO repo

File: Backend/WebAPI/Controllers/UsersController.cs [hunk 16]
```diff
@@ -65,6 +71,45 @@ public string Register([FromBody] RegisterRequest request)
...
+        /// <param name="request"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("api/users/login")]
+        public IHttpActionResult Login([FromBody] LoginRequest request)
+        {
+            IUserService _userService = new UserService();
+            IPasswordService _passwordService = new PasswordService();
+
+            using (var _db = new DatabaseContext())
```
Comment (alfredovargasjr) [inline Backend/WebAPI/Controllers/UsersController.cs]: This logic can be moved to the manager layer. If we feel like you can do it by the end of the milestone, go for it. If not, do it after so we can get functionality out of the way.

File: Backend/WebAPI/Controllers/UsersController.cs [hunk 17]
```diff
@@ -65,6 +71,45 @@ public string Register([FromBody] RegisterRequest request)
...
+            {
+                User user = _userService.GetUser(_db, request.email);
+                //PasswordReset ps = ps.
+                bool isPasswordMatched = _passwordService.VerifyPassword(request.password, user.PasswordHash, user.PasswordSalt);
+
+                //succesful login
+                if (isPasswordMatched == true)
```
Comment (alfredovargasjr) [inline Backend/WebAPI/Controllers/UsersController.cs]: This needs to return a token via a created session

File: Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs [hunk 18]
```diff
@@ -3,10 +3,24 @@ namespace DataAccessLayer.Migrations
...
+    public partial class initialcreate : DbMigration
     {
         public override void Up()
         {
+            CreateTable(
+                "dbo.Sessions",
+                c => new
+                    {
+                        Id = c.Guid(nullable: false),
+                        Token = c.String(nullable: false),
```
Comment (alfredovargasjr) [inline Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs]: Token should be a GUID

Comment (alfredovargasjr) [reply Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs]: Another team has implemented a Token Service

File: Backend/DataAccessLayer/Database/dbContext.cs (hunk 14 above)
Comment (cf2080) [reply Backend/DataAccessLayer/Database/dbContext.cs]: just for testing purposes when I was trying to connect to the database. I'll revert back to the original format

File: Backend/ServiceLayer/Services/PasswordService.cs (hunk 8 above)
Comment (cf2080) [reply Backend/ServiceLayer/Services/PasswordService.cs]: yes, functionality is in user repository class. I will use that one instead.

File: Backend/WebAPI/Controllers/UsersController.cs (hunk 17 above)
Comment (cf2080) [reply Backend/WebAPI/Controllers/UsersController.cs]: if statement now returns token through a created session, I believe.

File: Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs (hunk 1 above)
Comment (cf2080) [reply Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs]: accidentally deleted migration files when trying to connect to the database. @julianpoy

File: Backend/WebAPI/Controllers/UsersController.cs (hunk 16 above)
Comment (cf2080) [reply Backend/WebAPI/Controllers/UsersController.cs]: It's done. Can you check it? @alfredovargasjr

File: Backend/KFC-WebAPI/Controllers/UsersController.cs [hunk 19]
```diff
@@ -11,6 +11,7 @@
 using System.Net.Http;
 using System.Web.Http;
 using System.Web.Http.Cors;
+using DataAccessLayer.Repositories;
```
Comment (alfredovargasjr) [inline Backend/KFC-WebAPI/Controllers/UsersController.cs]: Why is this being called in the controller class?

File: Backend/DataAccessLayer/Models/LoginRequest.cs [hunk 20]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DataAccessLayer.Models
+{
```
Comment (alfredovargasjr) [inline Backend/DataAccessLayer/Models/LoginRequest.cs:2]: Any DTO's that are used for controllers should be in the Models folder inside the web-api project. KFC-WebApi/Models

File: Backend/KFC-WebAPI/Controllers/UsersController.cs [hunk 21]
```diff
@@ -63,6 +68,40 @@ public string Register([FromBody] RegisterRequest request)
...
+            {
+                if(loginM.LoginCheckUserDisabled())
+                {
+                    return Unauthorized();
+                }
+                else
+                {
+                    if(loginM.LoginCheckPassword(request.password))
+                    {
+                        return Ok(loginM.LoginAuthorized());
```
Comment (alfredovargasjr) [inline Backend/KFC-WebAPI/Controllers/UsersController.cs:92]: you should just return the token, not the entire session's object

File: Backend/KFC-WebAPI/Controllers/UsersController.cs [hunk 22]
```diff
@@ -63,6 +68,40 @@ public string Register([FromBody] RegisterRequest request)
...
+        /// <param name="request"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("api/users/login")]
+        public IHttpActionResult Login([FromBody] LoginRequest request)
+        {           
+            LoginManager loginM = new LoginManager();
+            if (loginM.LoginCheckUserExists(request.email) == false)
+            {
+                return NotFound();
```
Comment (alfredovargasjr) [inline Backend/KFC-WebAPI/Controllers/UsersController.cs]: Return some information so the error can be known. "Invalid username or password"

File: Backend/KFC-WebAPI/Scripts/index.d.ts [hunk 23]
```diff
@@ -0,0 +1,159 @@
+/**
+ * @fileoverview This file only declares the public portions of the API.
+ * It should not define internal pieces such as utils or modifier details.
+ *
+ * Original definitions by: edcarroll <https://github.com/edcarroll>, ggray <https://github.com/giladgray>, rhysd <https://rhysd.github.io>, joscha <https://github.com/joscha>, seckardt <https://github.com/seckardt>, marcfallows <https://github.com/marcfallows>
```
Comment (alfredovargasjr) [inline Backend/KFC-WebAPI/Scripts/index.d.ts]: This is not supposed to be in here

File: Backend/KFC-WebAPI/Scripts/README.md [hunk 24]
```diff
@@ -0,0 +1,219 @@
+<!-- IGNORE THE HTML BLOCK BELOW, THE INTERESTING PART IS AFTER IT -->
```
Comment (alfredovargasjr) [inline Backend/KFC-WebAPI/Scripts/README.md]: This has been removed by the bug branch. Accept the changes from the bug branch.

File: Backend/KFC-WebAPI/Scripts/index.d.ts (hunk 23 above)
Comment (alfredovargasjr) [reply Backend/KFC-WebAPI/Scripts/index.d.ts]: KFC-WebAPI/Scripts

**Review by alfredovargasjr — CHANGES_REQUESTED** [review]

File: Backend/KFC-WebAPI/Controllers/UsersController.cs (hunk 19 above)
Comment (cf2080) [reply Backend/KFC-WebAPI/Controllers/UsersController.cs]: I was using it when I had the majority of the logic in the class. I'm going to delete it.

File: Backend/DataAccessLayer/Models/LoginRequest.cs (hunk 20 above)
Comment (cf2080) [reply Backend/DataAccessLayer/Models/LoginRequest.cs:2]: Ok. I have a folder named DTOs in the DataAccessLayer, I'll delete it and move my DTO to the model's folder.

File: Backend/KFC-WebAPI/Scripts/README.md (hunk 24 above)
Comment (cf2080) [reply Backend/KFC-WebAPI/Scripts/README.md]: I merged the branch 2 days ago. I don't have anymore

File: Backend/KFC-WebAPI/Scripts/index.d.ts (hunk 23 above)
Comment (cf2080) [reply Backend/KFC-WebAPI/Scripts/index.d.ts]: I merged the branch 2 days ago. I don't have anymore

Comment (cf2080) [reply Backend/KFC-WebAPI/Scripts/index.d.ts]: with the bug branch

File: Backend/KFC-WebAPI/Scripts/README.md (hunk 24 above)
Comment (alfredovargasjr) [reply Backend/KFC-WebAPI/Scripts/README.md]: on that commit, you accepted some stuff but still seems like you kept old files.

File: Backend/KFC-WebAPI/Controllers/UsersController.cs (hunk 21 above)
Comment (cf2080) [reply Backend/KFC-WebAPI/Controllers/UsersController.cs:92]: the object does return a token

File: Backend/KFC-WebAPI/Controllers/UsersController.cs (hunk 22 above)
Comment (cf2080) [reply Backend/KFC-WebAPI/Controllers/UsersController.cs]: I changed it with **return content(httpstatuscode.notfound,"message");**

File: Backend/KFC-WebAPI/Controllers/UsersController.cs (hunk 21 above)
Comment (alfredovargasjr) [reply Backend/KFC-WebAPI/Controllers/UsersController.cs:92]: Got it. The var response was never used in the manager class, i was confused lol.

File: Backend/KFC-WebAPI/Content/bootstrap-grid.css [hunk 25]
```diff
@@ -0,0 +1,3719 @@
+/*!
```
Comment (alfredovargasjr) [inline Backend/KFC-WebAPI/Content/bootstrap-grid.css]: This also has to be removed, this is irrelevant for the web api project.

File: Backend/KFC-WebAPI/Content/bootstrap-reboot.css [hunk 26]
```diff
@@ -0,0 +1,331 @@
+/*!
```
Comment (alfredovargasjr) [inline Backend/KFC-WebAPI/Content/bootstrap-reboot.css]: Delete KFC-WebApi/content

**Review by alfredovargasjr — CHANGES_REQUESTED** [review]

File: .vs/ProjectSettings.json [hunk 27]
```diff
@@ -0,0 +1,3 @@
+{
+  "CurrentProjectSetting": null
```
Comment (julianpoy) [inline .vs/ProjectSettings.json]: This file should be removed

Comment (julianpoy) [reply .vs/ProjectSettings.json]: All other .vs files should be removed below

File: Backend/DataAccessLayer/App.config [hunk 28]
```diff
@@ -1,21 +1,20 @@
...
     </defaultconnectionfactory>
     <providers>
-      <provider invariantname="system.data.sqlclient" type="system.data.entity.sqlserver.sqlproviderservices, entityframework.sqlserver" />
+      <provider invariantname="system.data.sqlclient" type="system.data.entity.sqlserver.sqlproviderservices, entityframework.sqlserver"/>
     </providers>
   </entityframework>
   <connectionStrings>
-    <add name="ServerDB" providerName="System.Data.SqlClient"
-          connectionString="Data Source=localhost;Initial Catalog=SSO;Integrated Security=True" />
+    <add name="ServerDB" providerName="System.Data.SqlClient" connectionString="Data Source=localhost;Initial Catalog=SSO;Integrated Security=True"/>
```
Comment (julianpoy) [inline Backend/DataAccessLayer/App.config]: This file should not be changed

File: Backend/DataAccessLayer/DataAccessLayer.csproj [hunk 29]
```diff
@@ -49,15 +50,12 @@
...
-    <Compile Include="Migrations\201812040636053_create-table-user.cs" />
-    <Compile Include="Migrations\201812040636053_create-table-user.Designer.cs">
-      <DependentUpon>201812040636053_create-table-user.cs</DependentUpon>
-    </Compile>
-    <Compile Include="Migrations\201812040642468_create-table-session.cs" />
-    <Compile Include="Migrations\201812040642468_create-table-session.Designer.cs">
-      <DependentUpon>201812040642468_create-table-session.cs</DependentUpon>
+    <Compile Include="Migrations\201903062125096_initial-create.cs" />
+    <Compile Include="Migrations\201903062125096_initial-create.Designer.cs">
+      <DependentUpon>201903062125096_initial-create.cs</DependentUpon>
```
Comment (julianpoy) [inline Backend/DataAccessLayer/DataAccessLayer.csproj]: This migration should not be added, and the above migrations should not be removed

File: Backend/DataAccessLayer/DataAccessLayer.csproj [hunk 30]
```diff
@@ -75,13 +73,12 @@
     <Service Include="{508349B6-6B84-4DF5-91F0-309BEEBAD82D}" />
   </ItemGroup>
   <ItemGroup>
-    <EmbeddedResource Include="Migrations\201812040636053_create-table-user.resx">
-      <DependentUpon>201812040636053_create-table-user.cs</DependentUpon>
-    </EmbeddedResource>
-    <EmbeddedResource Include="Migrations\201812040642468_create-table-session.resx">
-      <DependentUpon>201812040642468_create-table-session.cs</DependentUpon>
+    <EmbeddedResource Include="Migrations\201903062125096_initial-create.resx">
+      <DependentUpon>201903062125096_initial-create.cs</DependentUpon>
```
Comment (julianpoy) [inline Backend/DataAccessLayer/DataAccessLayer.csproj]: Undo this

File: Backend/DataAccessLayer/DataAccessLayer.csproj [hunk 31]
```diff
@@ -75,13 +73,12 @@
...
     </EmbeddedResource>
   </ItemGroup>
-  <ItemGroup />
+  <ItemGroup>
+    <Folder Include="DTOs\" />
+  </ItemGroup>
```
Comment (julianpoy) [inline Backend/DataAccessLayer/DataAccessLayer.csproj]: Why this change?

File: Backend/DataAccessLayer/Database/DatabaseContext.cs [hunk 32]
```diff
@@ -16,10 +16,8 @@ public DatabaseContext()
         {
             this.Database.Connection.ConnectionString = "Data Source=localhost;Initial Catalog=SSO;Integrated Security=True";
         }
-
+        
         public DbSet<User> Users { get; set; }
         public DbSet<Session> Sessions { get; set; }
-
-
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Database/DatabaseContext.cs]: Undo all changes to this file

Comment (julianpoy) [reply Backend/DataAccessLayer/Database/DatabaseContext.cs]: All designer files below this one that have been removed need to be re-added

File: Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs (hunk 1 above)
Comment (julianpoy) [reply Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs]: Yeah, these need to be reversed before this can be resolved

File: Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs [hunk 33]
```diff
@@ -3,10 +3,24 @@ namespace DataAccessLayer.Migrations
...
+                        ExpiresAt = c.DateTime(nullable: false, precision: 7, storeType: "datetime2"),
+                        UpdatedAt = c.DateTime(nullable: false, precision: 7, storeType: "datetime2"),
+                        CreatedAt = c.DateTime(nullable: false, precision: 7, storeType: "datetime2"),
+                        UserId = c.Guid(nullable: false),
+                    })
+                .PrimaryKey(t => t.Id)
+                .ForeignKey("dbo.Users", t => t.UserId, cascadeDelete: true)
+                .Index(t => t.UserId);
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs]: Undo this block of green. It should all be unchanged, including the class name of this file.

File: Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs [hunk 34]
```diff
@@ -26,17 +40,21 @@ public override void Up()
...
+            
         }
         
         public override void Down()
         {
+            DropForeignKey("dbo.Sessions", "UserId", "dbo.Users");
+            DropIndex("dbo.Sessions", new[] { "UserId" });
             DropTable("dbo.Users");
-
+            DropTable("dbo.Sessions");
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs]: Undo this change

File: Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs [hunk 35]
```diff
@@ -26,17 +40,21 @@ public override void Up()
                         SecurityQ2Answer = c.String(),
                         SecurityQ3 = c.String(),
                         SecurityQ3Answer = c.String(),
+                        IncorrectPasswordCount = c.Int(nullable: false),
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs]: This needs to be in it's own new migration. You can't modify old migrations.

**Review by julianpoy — CHANGES_REQUESTED** [review]

File: Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs (hunk 1 above)
Comment (cf2080) [reply Backend/DataAccessLayer/Migrations/201903062125096_initial-create.cs]: files are now reversed

File: Backend/DataAccessLayer/DataAccessLayer.csproj (hunk 30 above)
Comment (cf2080) [reply Backend/DataAccessLayer/DataAccessLayer.csproj]: done

File: Backend/DataAccessLayer/DataAccessLayer.csproj (hunk 31 above)
Comment (cf2080) [reply Backend/DataAccessLayer/DataAccessLayer.csproj]: I created a folder to put the LoginRequest class there but decided to move it into the model class

**Comment by alfredovargasjr** [comment]
@cf2080 resolve conflicts

**Review by alfredovargasjr — DISMISSED** [review]

File: Frontend/src/views/Login.vue [hunk 36]
```diff
@@ -0,0 +1,60 @@
...
+            }
+        },
+        methods: {
+            login() {
+               axios.post('http://localhost:60461/api/users/login',
+               {input: this.input},
+               {
+                  'Access-Control-Allow-Origin': '*',
+                  'Access-Control-Allow-Credentials': true
+                })
```
Comment (julianpoy) [inline Frontend/src/views/Login.vue]: These headers should not be necessary.

**Review by julianpoy — DISMISSED** [review]

**Review by julianpoy — APPROVED** [review]
